engine/src/lib.rs

```diff
@@ -335,19 +421,26 @@ mod contract {
             &io,
         )
         .sdk_unwrap();
-        Engine::deploy_code_with_input(&mut engine, input, &mut Runtime)
-            .map(|res| res.try_to_vec().sdk_expect(errors::ERR_SERIALIZE))
+
+        Engine::deploy_code_with_input(&mut engine, input.clone(), &mut Runtime)
+            .map(|res| {
+                let output = res.try_to_vec().sdk_expect(errors::ERR_SERIALIZE);
```

Comment:
It's suboptimal that we end up serializing the output twice (once here and once in `sdk_process`. Maybe worth looking into removing that duplication of work, though the gas cost of serialization is probably not that high in most cases.

Reply:
I'm honestly not sure about how this is different from the base code:

```Engine::deploy_code_with_input(&mut engine, input, &mut Runtime)
            .map(|res| res.try_to_vec().sdk_expect(errors::ERR_SERIALIZE))
            .sdk_process();```